src/Mvc/Mvc.Core/src/Routing/ActionEndpointFactory.cs

```diff
@@ -88,7 +90,7 @@ public ActionEndpointFactory(RoutePatternTransformer routePatternTransformer, IE
                 endpoints.Add(builder.Build());
             }
 
-            if (action.AttributeRouteInfo == null)
+            if (action.AttributeRouteInfo?.Template == null)
```

Comment:
The code in the else block assumes presence of `AttributeRouteInfo` indicates a non-null template. The documentation appears to disagree: https://github.com/dotnet/aspnetcore/blob/master/src/Mvc/Mvc.Abstractions/src/Routing/AttributeRouteInfo.cs#L13-L16.

I don't think we produce an `AttributeRouteInfo` without a template configured, but this would appear to be the correct check.